### PR TITLE
fix C4244 warning

### PR DESCRIPTION
**Pull request by Mee-gu**
fix C4244 warning

### CODE REVIEW

File: cocos/2d/CCFastTMXLayer.cpp [hunk 1]
```diff
@@ -501,11 +501,12 @@ void TMXLayer::updateTotalQuads()
                 
                 float left, right, top, bottom, z;
                 
-                z = getVertexZForPos(Vec2(x, y));
-                auto iter = _indicesVertexZOffsets.find(z);
+                int zPos = getVertexZForPos(Vec2((float)x, (float)y));
+                z = (float)zPos;
```
Comment (minggo) [inline cocos/2d/CCFastTMXLayer.cpp:505]: it seems z is not used

Comment (Mee-gu) [reply cocos/2d/CCFastTMXLayer.cpp:505]: it used here 
https://github.com/cocos2d/cocos2d-x/blob/a500e7f0ea10c0d485269162022c15ab9cde9a4f/cocos/2d/CCFastTMXLayer.cpp#L527

File: cocos/2d/CCFontFreeType.cpp [hunk 2]
```diff
@@ -360,16 +360,17 @@ unsigned char* FontFreeType::getGlyphBitmap(uint64_t theChar, long &outWidth, lo
...
+                memset(blendImage, 0, imageSize);
 
                 auto px = outlineMinX - blendImageMinX;
                 auto py = blendImageMaxY - outlineMaxY;
                 for (int x = 0; x < outlineWidth; ++x)
                 {
                     for (int y = 0; y < outlineHeight; ++y)
                     {
-                        index = px + x + ((py + y) * blendWidth);
+                        index = (long)(px + x + ((py + y) * blendWidth));
```
Comment (minggo) [inline cocos/2d/CCFontFreeType.cpp]: does index need to be long?

Comment (Mee-gu) [reply cocos/2d/CCFontFreeType.cpp]: the integer type is enough, i'll fix it later.

**Review by minggo — APPROVED** [review]

**Pull request merged**